plugins/modules/cloudtrail_info.py

```diff
@@ -0,0 +1,244 @@
+            description: Specifies the date and time that CloudTrail last delivered a digest file to an account's Amazon S3 bucket.
+            type: str
+        latest_digest_delivery_error:
+            description: Displays any Amazon S3 error that CloudTrail encountered when attempting to deliver a digest file to the designated bucket.
+            type: str
+        resource_id:
+            description: Specifies the ARN of the resource.
+            type: str
+        tags:
+            description: Any tags assigned to the cloudtrail.
+            type: dict
+            returned: always
+            sample: "{ 'my_tag_key': 'my_tag_value' }"
+
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_filter_list
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+
+def get_trails(connection, module):
+    all_trails = []
+    next = False
+    try:
+        result = connection.list_trails()
+    except botocore.exceptions.ClientError as e:
+        module.fail_json_aws(e, msg="Failed to get the trails.")
+    while not all_trails or result.get("NextToken", ""):
+        all_trails.extend(list_cloud_trails(result))
+        if result.get("NextToken", ""):
+            next = True
+            try:
+                result = connection.list_trails(result["NextToken"])
+            except botocore.exceptions.ClientError as e:
+                module.fail_json_aws(e, msg="Failed to get the trails.")
+    if next:
+        all_trails.extend(list_cloud_trails(result))
+    return all_trails
+
+
+def list_cloud_trails(trail_dict):
+    return [x["TrailARN"] for x in trail_dict["Trails"]]
+
+
+def get_trail_detail(connection, module, trail_name_list=None, include_shadow_trails=True):
+    if not trail_name_list:
+        trail_name_list = get_trails(connection, module)
+    try:
+        result = connection.describe_trails(trailNameList=trail_name_list, includeShadowTrails=True)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json_aws(e, msg="Failed to get the trails.")
+    # Turn the boto3 result in to ansible_friendly_snaked_names
+    snaked_cloud_trail = []
+    for cloud_trail in result['trailList']:
+        try:
+            status_dict = connection.get_trail_status(Name=cloud_trail["TrailARN"], aws_retry=True)
+            cloud_trail.update(status_dict)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json_aws(e, msg="Failed to get the trail status")
+        try:
+            tag_list = connection.list_tags(ResourceIdList=[cloud_trail["TrailARN"]])
```

Comment:
I am not sure, but I think that when there are no tags associated to a trail we are not returning tags as ask empty dict.

Reply:
@alinabuzachis With the other keys, I see that when the value is not available, the key is not returned as part of the return payload, unless there is a default value set. IMO, `tags` should follow the same pattern. Please let me know why you think we should return {} for `tags` when the key doesn't have a value.

Reply:
The result returned by the _info module should be consistent with the one cloudtrail module returns. Please check that.